runtime/parachains/src/scheduler_polkadot/mod.rs

```diff
@@ -141,12 +155,23 @@ impl<T: scheduler::pallet::Config> AssignmentProvider<T> for Pallet<T> {
 	}
 
 	fn clear(core_idx: CoreIndex, assignment: Assignment) {
+		print("clear polkadot core");
+		print(core_idx.0);
 		let parachains_cores = <crate::scheduler_parachains::Pallet<T>>::session_core_count();
-		if (0..parachains_cores).contains(&core_idx.0) {
-			<crate::scheduler_parachains::Pallet<T>>::clear(core_idx, assignment)
-		} else {
-			let core_idx = CoreIndex(core_idx.0 - parachains_cores);
-			<crate::scheduler_parathreads::Pallet<T>>::clear(core_idx, assignment)
+		print("parachains_cores");
+		print(parachains_cores);
+
+		// We cannot rely on the CoreIndex to distinguish between parachains and -threads at this point
+		// as the number of parachains can change on new sessions.
+		// TODO: - check if this is correct.
```

Comment:
Eventually, I don't see why we need that clear() call at all. This might be the better solution. For the other question, I will need to check how this is used again.

Reply:
I agree. I also don't think clear() will be necessary at the end of the day but this is just to keep the structure similar to the old one for now.